models.js

```diff
@@ -0,0 +1,83 @@
+// ✅ Product 클래스
+export class Product {
+  constructor(
+    name,
+    description,
+    price,
+    tags = [],
+    images = [],
+    favoriteCount = 0
+  ) {
+    this.name = name;
+    this.description = description;
+    this.price = price;
+    this.tags = tags;
+    this.images = images;
+    this.favoriteCount = favoriteCount;
+  }
+
+  favorite() {
+    this.favoriteCount++;
+  }
+
+  toJSON() {
+    return {
+      name: this.name,
+      description: this.description,
+      price: this.price,
+      tags: this.tags,
+      images: this.images,
+      favoriteCount: this.favoriteCount,
+    };
+  }
+}
+
+// ✅ ElectronicProduct
+export class ElectronicProduct extends Product {
+  constructor(
+    name,
+    description,
+    price,
+    tags,
+    images,
+    favoriteCount,
+    manufacturer = 'Unknown'
+  ) {
+    super(name, description, price, tags, images, favoriteCount);
+    this.manufacturer = manufacturer;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      manufacturer: this.manufacturer,
+    };
+  }
+}
+
+// ✅ Article 클래스
+export class Article {
+  constructor(title, content, writer, likeCount = 0, image = null) {
```

Comment:
현재 `image`가 마지막에 있는데, 가이드 요구사항을 보면 Article의 기본 프로퍼티는 `title`, `content`, `writer`, `likeCount`이고 `image`는 API 요청에만 사용됩니다.
실제로 클래스 인스턴스를 만들 때 `image`를 자주 사용한다면 `likeCount`보다 앞에 두는 것도 고려해보세요.
```suggestion
  constructor(title, content, writer, image = null, likeCount = 0) {
```
이렇게 하면 `new Article('제목', '내용', '작성자', 'image.jpg')` 형태로 더 직관적으로 사용할 수 있습니다.